src/Refitter.Tests/Examples/TrimUnusedSchemaTests.cs

```diff
@@ -0,0 +1,214 @@
+﻿using FluentAssertions;
+
+using Refitter.Core;
+using Refitter.Tests.Build;
+
+using Xunit;
+
+namespace Refitter.Tests.Examples;
+
+public class TrimUnusedSchemaTests
+{
+    private const string OpenApiSpec = @"
+openapi: 3.0.1
+paths:
+  /v1/Warehouses:
+    post:
+      tags:
+        - Warehouses
+      operationId: CreateWarehouse
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Warehouse'
+      responses:
+        '201':
+          description: Created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Warehouse'
+        '400':
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+        '500':
+          description: Server Error
+components:
+  schemas:
+    Metadata:
+      type: object
+      properties:
+        createdAt:
+          type: string
+          format: date-time
+        createdBy:
+          type: string
+          nullable: true
+        lastModifiedAt:
+          type: string
+          format: date-time
+        lastModifiedBy:
+          type: string
+          nullable: true
+      additionalProperties: false
+    SomeComponent:
+      required:
+        - $type
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/Component'
+      properties:
+        $type:
+          type: string
+        typeId:
+          type: integer
+          format: int64
+      additionalProperties: false
+      discriminator:
+        propertyName: $type
+    SomeComponentState:
+      enum:
+        - Active
+        - Inactive
+        - Blocked
+        - Deleted
+      type: string
+    SomeComponentType:
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/Component'
+      properties:
+        state:
+          $ref: '#/components/schemas/SomeComponentState'
+        isBaseRole:
+          type: boolean
+        name:
+          type: string
+          nullable: true
+        numberingId:
+          type: string
+          nullable: true
+      additionalProperties: false
+    Component:
+      type: object
+      properties:
+        id:
+          type: integer
+          format: int64
+        metadata:
+          $ref: '#/components/schemas/Metadata'
+      additionalProperties: false
+    LoadingAddress:
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/SomeComponent'
+      properties:
+        info:
+          type: string
+          nullable: true
+      additionalProperties: false
+    Warehouse:
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/SomeComponent'
+      properties:
+        info:
+          type: string
+          nullable: true
+      additionalProperties: false
+    UserComponent:
```

Comment:
If it's not accessible from any request schema (request, header, parameters, response) it will be trimmed away.

OpenApi doesn't have "inheritance", so any Schema that _might_ be returned, should be in an AnyOf response type, thus won't be removed.